examples/front.kk

```diff
@@ -0,0 +1,97 @@
+/*----------------------------------------------------------------------------
+   Copyright 2024, Koka-Community Authors
+
+   Licensed under the MIT License ("The License"). You may not
+   use this file except in compliance with the License. A copy of the License
+   can be found in the LICENSE file at the root of this distribution.
+----------------------------------------------------------------------------*/
+
+import html/html
+
+import html/layout
+
+// todo: this deserves a PR to go into std/core?
+fun do(action : () -> e a) : e a
+  action()
+
+value struct cnode
+  swarm : string
+  ip : string
+  ncores : int
+  avail-ram : int
+  total-ram : int
+
+effect val cnodes-info : list<cnode>
+
+fun frontpage(): <pure,cnodes-info> component
+  val page-title = "The Front Page"
+
+  val template = web-tmpl-basic(page-head = fn(layout) {
+    page-head(web-tmpl-basic)(layout)
+    link("stylesheet", "text/css", "/front.css")
+  })
+
+  with super <- defaultpage()
+  with layout <- named handler
+    fun set-body-begin-scripts(b)
+      super.set-body-begin-scripts(b)
+    fun set-body-end-scripts(e)
+      super.set-body-end-scripts(e)
+    fun set-body-content(c)
+      super.set-body-content(c)
+    fun set-page-title(t)
+      super.set-page-title(t)
```

Comment:
With the way you designed this we can probably get rid of the `set-*` methods. Which would make this somewhat nicer.

Reply:
yeah, agree.